### PR TITLE
Connection between contributor and search page

### LINKED ISSUE #45: Feature: connecting search and contributors page
### Description:
Currently, there's a disconnect between the search and contributors pages. After a successful search, users should be able to seamlessly see contributors of the selected repository.

### Tips for the issue:
- Implement a button or link on the search results page to access the contributors.
- Ensure that clicking the button provides users with information about the contributors of the selected repository.

### To do:
- [x] Ask us to assign the issue
- [x] Once the issue is assigned, you can start working on it
- [x] Create a PR

### Resource:
- [Material 3 Design](https://docs.flutter.dev/ui/widgets/material) 
- [Flutter Navigation and routing docs](https://docs.flutter.dev/ui/navigation) 

### Notes:
The task is **assigned to contributors on a first-come, first-serve basis**, and the **contributor has to report progress every 3 days** to ensure active development on the tasks.

### PR DESCRIPTION
**Pull request by Lomesh21**
#45 feature added
[Feature: connecting search and contributors page](https://github.com/IEEE-VIT/hackfest-flutter/issues/45)

## Changes made
Contributors button added in repo card
Functionality of button is also implemented

## For each repo the contributors of that repo is displayed on contributors page

Resolves #45

### CODE REVIEW

File: lib/widgets/new_repo_card.dart
Comment (dk-a-dev) [inline lib/widgets/new_repo_card.dart:1]: #### Comment: Good job adding and connecting pages
#### Changes and Fixes:
- "visit" button is not working and has a unhandled exception
image

- the buttons don't look alike, shape, colour scheme make sure to fix this  and some space b/w them , and align them properly

File: lib/screens/contributors.dart [hunk 1]
```diff
@@ -22,7 +22,7 @@ class _ContributorsState extends State<Contributors>
   @override
   void initState() {
     super.initState();
-    getContributors(username: 'IEEE-VIT', repository: 'hacktoberfest-flutter')
+    getContributors(repository: widget.repoName)
```
Comment (dk-a-dev) [inline lib/screens/contributors.dart:25]: @Lomesh21 The primary aim of the devs of the repo was to retain the contributors of this project in contributors page,
so what we can do is shift that aim to the About Us page, with one click people can see the contributors of this repo and also visit, ik this is out of scope of this issue, but if u want to work on this I can create a separate issue or extend scope of this one le me knw

File: lib/screens/home_page.dart
Comment (dk-a-dev) [inline lib/screens/home_page.dart:1]: Import on line 2 seems unused check plz

File: lib/widgets/new_repo_card.dart [hunk 2]
```diff
@@ -1,8 +1,11 @@
+import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_slimy_card/flutter_slimy_card.dart';
 import 'package:hacktoberfest_flutter/shared/colors.dart';
 import 'package:url_launcher/url_launcher.dart';
 
+import '../screens/contributors.dart';
```
Comment (dk-a-dev) [inline lib/widgets/new_repo_card.dart:7]: convert it to 
> import 'package:hacktoberfest_flutter/screens/contributors.dart';

**Review by dk-a-dev — CHANGES_REQUESTED** [review]
Review:
- some fixes mentioned
- reply on whatever I asked

File: lib/screens/contributors.dart (hunk 1 above)
Comment (Lomesh21) [reply lib/screens/contributors.dart:25]: Can you create new issue with more descriptive, I am happy to work on this.

File: lib/screens/home_page.dart
Comment (Lomesh21) [reply lib/screens/home_page.dart:1]: Yep I'll fix it

**Comment by dk-a-dev** [comment]
Let me know once u r done with changes
@Lomesh21

File: lib/widgets/new_repo_card.dart (hunk 2 above)
Comment (Lomesh21) [reply lib/widgets/new_repo_card.dart:7]: Gotcha

File: lib/screens/contributors.dart (hunk 1 above)
Comment (dk-a-dev) [reply lib/screens/contributors.dart:25]: @Lomesh21 ok lets merge this one first and then I will create an issue for this

File: lib/widgets/new_repo_card.dart
Comment (Lomesh21) [reply lib/widgets/new_repo_card.dart:1]: Visit button is working fine in my emulator,


https://github.com/IEEE-VIT/hackfest-flutter/assets/94422862/648cfe98-4b3e-4544-bb49-8aae6274fac1


I will check what wrong on my side but please check on your side also

**Comment by Lomesh21** [comment]
Hey @dk-a-dev, I have fixed the issues as much as possible from my side, the visit issue you faced is not on my side, I have checked all the dependencies and all but I didnt find anything related to it

**Review by dk-a-dev — CHANGES_REQUESTED** [review]
Fixes:
- overflow error( try to wrap up those buttons a/c to user phone width and card size
image
image
Rest everything seems fine
I am still checking that "visit button problem"
Its on master ig, I will look into that
please fix overflow problem

**Comment by Lomesh21** [comment]
@dk-a-dev  I tried to make buttons responsive but the text size may cause overflow again and again if mobile width minimizes. That's why i made them in column way instead of row way, please let me know what are your suggestions.

**Comment by dk-a-dev** [comment]
Dw it's fine , UI revamp issue will take care of that, ur pr is under review

**Review by dk-a-dev — APPROVED** [review]
Good Job buddy
- That visit problem is easy to sort, I fill fix it myself
Anyways I will be adding more issues to inc functionality in contributors page and also in about us tab

**Pull request merged**